### PR TITLE
Add archive taps as a single jar file

### PR DESCRIPTION
**Pull request by sophia-guo**
Tap files per platforms keep as before.

Parts of https://github.com/adoptium/aqa-tests/issues/5242

### CODE REVIEW

**Comment by sophia-guo** [comment]
https://ci.adoptium.net/job/build-scripts/job/openjdk21-pipeline/275/artifact/
Screenshot 2024-05-08 at 1 36 06 PM

**Review by smlambert — APPROVED** [review]
Thanks, this is superior to the current approach which is use the Jenkins GUI to download the folder as a single zip file.
aqavitTaps

**Comment by sophia-guo** [comment]
This is only for upstream pipeline job (e.g. **openjdk**-pipeline), which zips all tap files under different directory of `target/${config.TARGET_OS}/${config.ARCHITECTURE}/${config.VARIANT}/AQAvitTaps/`. The tar preserves the directory structure.

Nothing is changed to build job, the zip file for build job can still use the Jenkins GUI to download .

File: pipelines/build/common/build_base_file.groovy [hunk 1]
```diff
@@ -968,6 +968,16 @@ class Builder implements Serializable {
                                             throw new Exception("[ERROR] Archive artifact timeout (${pipelineTimeouts.ARCHIVE_ARTIFACTS_TIMEOUT} HOURS) for ${downstreamJobName}has been reached. Exiting...")
                                         }
 
+                                        //Archive tap files as a single tar file
```
Comment (karianna) [inline pipelines/build/common/build_base_file.groovy:971]:
```suggestion
                                        // Archive tap files as a single tar file
```

**Review by karianna — APPROVED** [review]

File: pipelines/build/common/build_base_file.groovy [hunk 2]
```diff
@@ -968,6 +968,16 @@ class Builder implements Serializable {
                                             throw new Exception("[ERROR] Archive artifact timeout (${pipelineTimeouts.ARCHIVE_ARTIFACTS_TIMEOUT} HOURS) for ${downstreamJobName}has been reached. Exiting...")
                                         }
 
+                                        // Archive tap files as a single tar file
+                                        context.sh "find . -type f -name '*.tap' -exec tar -czf AQAvitTapFiles.tar.gz {} + "
```
Comment (keithc-ca) [inline pipelines/build/common/build_base_file.groovy:972]: Given a sufficient number of `*.tap` files, `find` will run `tar` more than once, the result being that the archive only contains the last batch of `*.tap` files.

Comment (sophia-guo) [reply pipelines/build/common/build_base_file.groovy:972]: Thanks @keithc-ca . You are correct. Will update

**Pull request merged**